backend/src/email/middlewares/email-transactional.middleware.ts

```diff
@@ -41,6 +45,27 @@ async function sendMessage(
   }
 }
 
+const rateLimit = expressRateLimit({
+  store: new RedisStore({
+    prefix: 'email-transactional:',
+    client: RedisService.rateLimitClient,
+    expiry: config.get('transactionalEmail.window'),
+  }),
+  keyGenerator: () => 'global',
+  windowMs: config.get('transactionalEmail.window') * 1000,
+  max: config.get('transactionalEmail.rate'),
+  draft_polli_ratelimit_headers: true,
+  message: {
+    status: 429,
+    message: 'Too many requests. Please try again later.',
+  },
+  handler: (_: Request, res: Response) => {
+    logger.warn({ message: 'Rate limited request to send transactional email' })
```

Comment:
Would be good to log information about the user to provide more context.

Reply:
I've logged user IDs. I think this is enough since we can just look up more details from the DB.